meerk40t/gui/scene/scenewidgets.py

```diff
@@ -1026,12 +1027,37 @@ def tool_rotate(self, position, dx, dy, event=0):
 
             # Improved code by tatarize to establish rotation angle
             current_angle = Point.angle(position[:2], (self.rotate_cx, self.rotate_cy))
+            # boost the angle to allow for simpler bigger movements
+            if self.key_shift_pressed:
+                current_angle *= 2
+            elif self.key_control_pressed:
+                current_angle *= 4
```

Comment:
The code is a lot better usage-wise without this angle-boost. The transitions from 1 deg angle to stepped angles doesn't spin weirdly as the previous current_angle is unceremoniously doubled or quadrupled, and it preserves the angle between the center of the object and the mouse is pretty much always constant.

Reply:
Done